pkg/liqo-controller-manager/external-network/utils/getters.go

```diff
@@ -124,3 +126,25 @@ func KindToResource(kind string) string {
 	// lowercased and pluralized
 	return strings.ToLower(kind) + "s"
 }
+
+// ResourceToKind returns the kind name for a given resource.
+func ResourceToKind(gvr schema.GroupVersionResource, config *rest.Config) (string, error) {
+	dClient, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return "", err
+	}
+
+	res, err := dClient.ServerResourcesForGroupVersion(gvr.GroupVersion().String())
+	if err != nil {
+		return "", err
+	}
+
+	for i := range res.APIResources {
+		k := res.APIResources[i].Kind
+		if gvr.Resource == KindToResource(k) {
```

Comment:
https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/restmapper.go#L126
Can you use this function? We already discussed about it but maybe we forgot to change it.